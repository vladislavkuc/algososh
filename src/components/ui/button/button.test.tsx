import renderer from 'react-test-renderer'
const { render, screen, fireEvent } = jest.requireActual('@testing-library/react');

const { Button } = jest.requireActual('./button');

it('Кнопка с текстом отрисовывается корректно', () => {
  const tree = renderer
    .create(<Button text='Кнопка'/>)
    .toJSON();
    expect(tree).toMatchSnapshot();
});

it('Кнопка без текста отрисовывается корректно', () => {
  const tree = renderer
    .create(<Button/>)
    .toJSON();
    expect(tree).toMatchSnapshot();
});

it('Заблокированная кнопка с текстом отрисовывается корректно', () => {
  const tree = renderer
    .create(<Button text='Кнопка' disabled={true}/>)
    .toJSON();
    expect(tree).toMatchSnapshot();
});

it('Кнопка с текстом и индикацией загрузки отрисовывается корректно', () => {
  const tree = renderer
    .create(<Button text='Кнопка' isLoader={false}/>)
    .toJSON();
    expect(tree).toMatchSnapshot();
});

it('Нажатие на кнопку вызывает корректный callback', () => {
  window.alert = jest.fn();

  render(<Button text='Вызвать alert' onClick={() => { alert('alert вызван успешно')}}/>)

  const button = screen.getByText("Вызвать alert");

  fireEvent.click(button);

  expect(window.alert).toHaveBeenCalledWith('alert вызван успешно');
});

export {};
