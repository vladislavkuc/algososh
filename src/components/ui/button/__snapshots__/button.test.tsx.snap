// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Заблокированная кнопка с текстом отрисовывается корректно 1`] = `
<button
  className="text text_type_button text_color_primary undefined undefined false "
  disabled={true}
  type="button"
>
  <p
    className="text undefined"
  >
    Кнопка
  </p>
</button>
`;

exports[`Кнопка без текста отрисовывается корректно 1`] = `
<button
  className="text text_type_button text_color_primary undefined undefined false "
  type="button"
>
  <p
    className="text undefined"
  />
</button>
`;

exports[`Кнопка с текстом и индикацией загрузки отрисовывается корректно 1`] = `
<button
  className="text text_type_button text_color_primary undefined undefined false "
  type="button"
>
  <p
    className="text undefined"
  >
    Кнопка
  </p>
</button>
`;

exports[`Кнопка с текстом отрисовывается корректно 1`] = `
<button
  className="text text_type_button text_color_primary undefined undefined false "
  type="button"
>
  <p
    className="text undefined"
  >
    Кнопка
  </p>
</button>
`;
